Here I continue with some preprocessing after the processing done in ERS_main. I also perform a logistic regression without accounting for categorical data with a prediction score above 84%.

In [293]:
import pandas as pd
import numpy as np

In [150]:
pwd

'/Users/gabriel/Documents/data'

In [151]:
import os 
path = ('/Users/gabriel/Documents/data')
os.chdir(path)

In [152]:
df = pd.read_csv('Main_First_Preprocessing.csv')

/anaconda3/envs/py3-TF2.0/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3156: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [153]:
df.columns

Index(['A0', 'A1', 'A2', 'A3', 'PG1', 'PE1', 'PL1', 'PM1', 'PN1', 'PO1', 'C1',
       'C2', 'D0', 'D1', 'D2', 'D3', 'D4', 'D5', 'D6', 'QTI', 'V0', 'V1',
       'V10', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'TP', 'GWP'],
      dtype='object')

In [154]:
cat_first = ['A0', 'V8','PM1','C1','C2','A1', 'A2', 'A3', 'PG1', 'PE1', 'PL1' , 'PN1', 'PO1',
        'D0', 'D1', 'D2', 'D3', 'D4', 'D5', 'D6', 'QTI', 'V0', 'V1',
       'V10', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7' , 'V9', 'TP', 'GWP']

In [155]:
pd.options.display.max_columns = None #Display all columns

In [156]:
#Categorical data first
df = df[cat_first]
df.head(5)

A0       V8 PM1             C1   C2    A1    A2    A3  PG1  \
0       West Midlands  At Home   M  Comprehensive    L  54.0  45.0  62.0   54   
1     Yorks and Humbs  At Home   S  Comprehensive  NaN  80.0  71.0  73.0   28   
2  South-East England  At Home   S  Comprehensive  NaN  25.0  43.0  50.0   25   
3               Wales  At Home   S  Comprehensive  NaN  51.0  56.0  63.0   26   
4         East Anglia  At Home   M  Comprehensive  NaN  21.0  20.0  34.0   82   

   PE1  PL1  PN1  PO1  D0     D1   D2   D3    D4    D5     D6  QTI    V0  \
0  2.0  1.0    6  9.0   0  708.0  0.0  0.0  21.0  42.0  155.0   30  51.0   
1  0.0  1.0    5  9.0   0  746.0  0.0  0.0  13.0  39.0  112.0   15  41.0   
2  0.0  1.0    5  9.0   0  809.0  0.0  0.0   7.0  37.0  106.0   15  74.0   
3  0.0  1.0    4  9.0   0  708.0  0.0  0.0  17.0  42.0  119.0   28  62.0   
4  2.0  1.0    2  9.0   0  816.0  0.0  0.0  19.0  39.0  111.0   30  32.0   

     V1  V10    V2    V3      V4     V5  V6       V7   V9      TP     GWP  
0  63.0    1  10.0  10.0  6345.0  250.0   0   4000.0  610   39.49   382.0  
1  21.0    1  10.0   5.0  4000.0  500.0   0   8000.0  229  229.33  1176.0  
2  29.0    1   8.0   9.0  2800.0  150.0   0   8000.0   76  286.27  2152.0  
3  42.0    1  10.0   6.0  6700.0  250.0   0   8000.0  789  246.94  1878.0  
4  19.0    1  10.0   2.0  7795.0    0.0   0  10000.0  761   31.43   412.0

# Categorical data

In [157]:
categorical_df = df.iloc[:,:5]
categorical_df.head(3)

A0       V8 PM1             C1   C2
0       West Midlands  At Home   M  Comprehensive    L
1     Yorks and Humbs  At Home   S  Comprehensive  NaN
2  South-East England  At Home   S  Comprehensive  NaN

In [158]:
categorical_df.shape

(647206, 5)

# Getting Dummies

In [272]:
a0 = categorical_df.drop(['V8','PM1','C1','C2'],axis = 1)
dum_a0 = pd.get_dummies(a0,prefix = None)
dum_a0.head()

A0_East Anglia  A0_East Midlands  A0_Greater London  A0_Non-Mainland UK   \
0               0                 0                  0                    0   
1               0                 0                  0                    0   
2               0                 0                  0                    0   
3               0                 0                  0                    0   
4               1                 0                  0                    0   

   A0_North-East England  A0_North-West England  A0_Others  A0_Scotland  \
0                      0                      0          0            0   
1                      0                      0          0            0   
2                      0                      0          0            0   
3                      0                      0          0            0   
4                      0                      0          0            0   

   A0_South-East England  A0_South-West England  A0_Wales  A0_West Midlands  \
0                      0                      0         0                 1   
1                      0                      0         0                 0   
2                      1                      0         0                 0   
3                      0                      0         1                 0   
4                      0                      0         0                 0   

   A0_Yorks and Humbs  
0                   0  
1                   1  
2                   0  
3                   0  
4                   0

In [265]:
v8 = categorical_df.drop(['A0','PM1','C1','C2'],axis = 1)
dum_v8 = pd.get_dummies(v8)
dum_v8.head()

V8_At Home  V8_Car Park  V8_Garage  V8_Secure Carpark
0           1            0          0                  0
1           1            0          0                  0
2           1            0          0                  0
3           1            0          0                  0
4           1            0          0                  0

In [273]:
pm1 = categorical_df.drop(['A0','V8','C1','C2'],axis = 1)
dum_pm1 = pd.get_dummies(pm1)
pm1.head()

PM1
0   M
1   S
2   S
3   S
4   M

In [274]:
c1 = categorical_df.drop(['A0','PM1','V8','C2'],axis = 1)
dum_c1 = pd.get_dummies(c1)
dum_c1.head()

C1_Comprehensive  C1_Third Party Fire and Theft  C1_Third Party Only
0                 1                              0                    0
1                 1                              0                    0
2                 1                              0                    0
3                 1                              0                    0
4                 1                              0                    0

In [277]:
c2 = categorical_df.drop(['A0','PM1','C1','V8'],axis = 1)
dum_c2 = pd.get_dummies(c2)
dum_c2.head()

C2_A  C2_B  C2_L  C2_M  C2_N  C2_O
0     0     0     1     0     0     0
1     0     0     0     0     0     0
2     0     0     0     0     0     0
3     0     0     0     0     0     0
4     0     0     0     0     0     0

# Scaling Numerical Data

In [159]:
numerical_df = df.iloc[:,5:]
numerical_df.head(3)

A1    A2    A3  PG1  PE1  PL1  PN1  PO1  D0     D1   D2   D3    D4    D5  \
0  54.0  45.0  62.0   54  2.0  1.0    6  9.0   0  708.0  0.0  0.0  21.0  42.0   
1  80.0  71.0  73.0   28  0.0  1.0    5  9.0   0  746.0  0.0  0.0  13.0  39.0   
2  25.0  43.0  50.0   25  0.0  1.0    5  9.0   0  809.0  0.0  0.0   7.0  37.0   

      D6  QTI    V0    V1  V10    V2    V3      V4     V5  V6      V7   V9  \
0  155.0   30  51.0  63.0    1  10.0  10.0  6345.0  250.0   0  4000.0  610   
1  112.0   15  41.0  21.0    1  10.0   5.0  4000.0  500.0   0  8000.0  229   
2  106.0   15  74.0  29.0    1   8.0   9.0  2800.0  150.0   0  8000.0   76   

       TP     GWP  
0   39.49   382.0  
1  229.33  1176.0  
2  286.27  2152.0

In [180]:
numerical_df.columns

Index(['A1', 'A2', 'A3', 'PG1', 'PE1', 'PL1', 'PN1', 'PO1', 'D0', 'D1', 'D2',
       'D3', 'D4', 'D5', 'D6', 'QTI', 'V0', 'V1', 'V10', 'V2', 'V3', 'V4',
       'V5', 'V6', 'V7', 'V9', 'TP', 'GWP'],
      dtype='object')

In [160]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

In [161]:
scaler.fit(numerical_df)

StandardScaler()

In [162]:
scaled_numerical = scaler.transform(numerical_df)

In [163]:
scaled_numerical

array([[ 0.54057351, -0.20660206,  0.5243613 , ..., -0.30820443,
        -0.59374226, -0.6521107 ],
       [ 1.93122901,  1.03337497,  1.05287486, ..., -0.73128221,
         0.93891782,  0.23802213],
       [-1.01054224, -0.3019849 , -0.05219895, ..., -0.90117958,
         1.39861897,  1.33219044],
       ...,
       [        nan,         nan,         nan, ..., -0.39703966,
         0.13924451,  1.13600248],
       [        nan,         nan,         nan, ..., -0.83899492,
         0.4227834 ,  0.82434388],
       [        nan,         nan,         nan, ..., -0.98557305,
        -0.27815185,  1.73353497]])

In [178]:
scaled_numerical.shape

(647206, 28)

In [181]:
scaled_numerical_final = pd.DataFrame(scaled_numerical, columns = ['A1', 'A2', 'A3', 'PG1', 'PE1', 'PL1', 'PN1', 'PO1', 'D0', 'D1', 'D2',
       'D3', 'D4', 'D5', 'D6', 'QTI', 'V0', 'V1', 'V10', 'V2', 'V3', 'V4',
       'V5', 'V6', 'V7', 'V9', 'TP', 'GWP'])

In [182]:
scaled_numerical_final.head()

A1        A2        A3       PG1       PE1       PL1       PN1  \
0  0.540574 -0.206602  0.524361  0.432767  0.486076  0.023917 -0.310123   
1  1.931229  1.033375  1.052875 -1.233143 -1.445821  0.023917 -0.553836   
2 -1.010542 -0.301985 -0.052199 -1.425363 -1.445821  0.023917 -0.553836   
3  0.380113  0.318004  0.572408 -1.361290 -1.445821  0.023917 -0.797549   
4 -1.224489 -1.398888 -0.820946  2.226823  0.486076  0.023917 -1.284974   

        PO1        D0        D1        D2        D3        D4        D5  \
0  0.149802 -0.180638 -1.305960 -0.104297 -0.043828  0.646512 -0.035291   
1  0.149802 -0.180638 -0.647928 -0.104297 -0.043828 -1.405344 -0.701139   
2  0.149802 -0.180638  0.443019 -0.104297 -0.043828 -2.944236 -1.145037   
3  0.149802 -0.180638 -1.305960 -0.104297 -0.043828 -0.379416 -0.035291   
4  0.149802 -0.180638  0.564235 -0.104297 -0.043828  0.133548 -0.701139   

         D6       QTI        V0        V1       V10        V2        V3  \
0  2.344680  0.269630 -0.150814  2.091244 -0.056761  0.269289  0.178104   
1 -0.489017  0.052746 -0.640506 -0.941047 -0.056761  0.269289 -0.188684   
2 -0.884417  0.052746  0.975477 -0.363468 -0.056761 -1.996511  0.104747   
3 -0.027718  0.240712  0.387847  0.575098 -0.056761  0.269289 -0.115327   
4 -0.554917  0.269630 -1.081228 -1.085442 -0.056761  0.269289 -0.408757   

         V4        V5        V6        V7        V9        TP       GWP  
0 -0.382676  0.155328 -0.032119 -0.558132 -0.308204 -0.593742 -0.652111  
1 -0.718096  1.751069 -0.032119 -0.074237 -0.731282  0.938918  0.238022  
2 -0.889739 -0.482968 -0.032119 -0.074237 -0.901180  1.398619  1.332190  
3 -0.331899  0.155328 -0.032119 -0.074237 -0.109436  1.081091  1.025016  
4 -0.175274 -1.440413 -0.032119  0.167710 -0.140528 -0.658814 -0.618478

In [217]:
scaled_nonan = scaled_numerical_final.fillna(scaled_numerical_final.mean())

In [218]:
scaled_nonan.isna().sum() #No nan values

A1     0
A2     0
A3     0
PG1    0
PE1    0
PL1    0
PN1    0
PO1    0
D0     0
D1     0
D2     0
D3     0
D4     0
D5     0
D6     0
QTI    0
V0     0
V1     0
V10    0
V2     0
V3     0
V4     0
V5     0
V6     0
V7     0
V9     0
TP     0
GWP    0
dtype: int64

# Setting Targets

In [220]:
median = scaled_nonan['GWP'].median()
median

-0.30906203116674097

In [221]:
targets = np.where(scaled_nonan['GWP']>median,1,0) #Spliting targets into 0s and 1s depending on the median 
targets

array([0, 1, 1, ..., 1, 1, 1])

In [222]:
(np.sum(targets)/targets.shape[0]).round(2) #To find whether the the split aprox 50-50

0.5

In [224]:
targets = pd.DataFrame(targets, columns = ['Targets'])
targets.head(3)

Targets
0        0
1        1
2        1

In [225]:
numerical_with_gwp = pd.concat([scaled_nonan,targets],axis = 1)
numerical_with_gwp.head()

A1        A2        A3       PG1       PE1       PL1       PN1  \
0  0.540574 -0.206602  0.524361  0.432767  0.486076  0.023917 -0.310123   
1  1.931229  1.033375  1.052875 -1.233143 -1.445821  0.023917 -0.553836   
2 -1.010542 -0.301985 -0.052199 -1.425363 -1.445821  0.023917 -0.553836   
3  0.380113  0.318004  0.572408 -1.361290 -1.445821  0.023917 -0.797549   
4 -1.224489 -1.398888 -0.820946  2.226823  0.486076  0.023917 -1.284974   

        PO1        D0        D1        D2        D3        D4        D5  \
0  0.149802 -0.180638 -1.305960 -0.104297 -0.043828  0.646512 -0.035291   
1  0.149802 -0.180638 -0.647928 -0.104297 -0.043828 -1.405344 -0.701139   
2  0.149802 -0.180638  0.443019 -0.104297 -0.043828 -2.944236 -1.145037   
3  0.149802 -0.180638 -1.305960 -0.104297 -0.043828 -0.379416 -0.035291   
4  0.149802 -0.180638  0.564235 -0.104297 -0.043828  0.133548 -0.701139   

         D6       QTI        V0        V1       V10        V2        V3  \
0  2.344680  0.269630 -0.150814  2.091244 -0.056761  0.269289  0.178104   
1 -0.489017  0.052746 -0.640506 -0.941047 -0.056761  0.269289 -0.188684   
2 -0.884417  0.052746  0.975477 -0.363468 -0.056761 -1.996511  0.104747   
3 -0.027718  0.240712  0.387847  0.575098 -0.056761  0.269289 -0.115327   
4 -0.554917  0.269630 -1.081228 -1.085442 -0.056761  0.269289 -0.408757   

         V4        V5        V6        V7        V9        TP       GWP  \
0 -0.382676  0.155328 -0.032119 -0.558132 -0.308204 -0.593742 -0.652111   
1 -0.718096  1.751069 -0.032119 -0.074237 -0.731282  0.938918  0.238022   
2 -0.889739 -0.482968 -0.032119 -0.074237 -0.901180  1.398619  1.332190   
3 -0.331899  0.155328 -0.032119 -0.074237 -0.109436  1.081091  1.025016   
4 -0.175274 -1.440413 -0.032119  0.167710 -0.140528 -0.658814 -0.618478   

   Targets  
0        0  
1        1  
2        1  
3        1  
4        0

In [226]:
numerical_with_targets = numerical_with_gwp.drop('GWP',axis = 1)
numerical_with_targets.head()

A1        A2        A3       PG1       PE1       PL1       PN1  \
0  0.540574 -0.206602  0.524361  0.432767  0.486076  0.023917 -0.310123   
1  1.931229  1.033375  1.052875 -1.233143 -1.445821  0.023917 -0.553836   
2 -1.010542 -0.301985 -0.052199 -1.425363 -1.445821  0.023917 -0.553836   
3  0.380113  0.318004  0.572408 -1.361290 -1.445821  0.023917 -0.797549   
4 -1.224489 -1.398888 -0.820946  2.226823  0.486076  0.023917 -1.284974   

        PO1        D0        D1        D2        D3        D4        D5  \
0  0.149802 -0.180638 -1.305960 -0.104297 -0.043828  0.646512 -0.035291   
1  0.149802 -0.180638 -0.647928 -0.104297 -0.043828 -1.405344 -0.701139   
2  0.149802 -0.180638  0.443019 -0.104297 -0.043828 -2.944236 -1.145037   
3  0.149802 -0.180638 -1.305960 -0.104297 -0.043828 -0.379416 -0.035291   
4  0.149802 -0.180638  0.564235 -0.104297 -0.043828  0.133548 -0.701139   

         D6       QTI        V0        V1       V10        V2        V3  \
0  2.344680  0.269630 -0.150814  2.091244 -0.056761  0.269289  0.178104   
1 -0.489017  0.052746 -0.640506 -0.941047 -0.056761  0.269289 -0.188684   
2 -0.884417  0.052746  0.975477 -0.363468 -0.056761 -1.996511  0.104747   
3 -0.027718  0.240712  0.387847  0.575098 -0.056761  0.269289 -0.115327   
4 -0.554917  0.269630 -1.081228 -1.085442 -0.056761  0.269289 -0.408757   

         V4        V5        V6        V7        V9        TP  Targets  
0 -0.382676  0.155328 -0.032119 -0.558132 -0.308204 -0.593742        0  
1 -0.718096  1.751069 -0.032119 -0.074237 -0.731282  0.938918        1  
2 -0.889739 -0.482968 -0.032119 -0.074237 -0.901180  1.398619        1  
3 -0.331899  0.155328 -0.032119 -0.074237 -0.109436  1.081091        1  
4 -0.175274 -1.440413 -0.032119  0.167710 -0.140528 -0.658814        0

In [227]:
#Checkpoint
numerical_df4 = numerical_with_targets.copy() 

In [228]:
numerical_df4.head(3)

A1        A2        A3       PG1       PE1       PL1       PN1  \
0  0.540574 -0.206602  0.524361  0.432767  0.486076  0.023917 -0.310123   
1  1.931229  1.033375  1.052875 -1.233143 -1.445821  0.023917 -0.553836   
2 -1.010542 -0.301985 -0.052199 -1.425363 -1.445821  0.023917 -0.553836   

        PO1        D0        D1        D2        D3        D4        D5  \
0  0.149802 -0.180638 -1.305960 -0.104297 -0.043828  0.646512 -0.035291   
1  0.149802 -0.180638 -0.647928 -0.104297 -0.043828 -1.405344 -0.701139   
2  0.149802 -0.180638  0.443019 -0.104297 -0.043828 -2.944236 -1.145037   

         D6       QTI        V0        V1       V10        V2        V3  \
0  2.344680  0.269630 -0.150814  2.091244 -0.056761  0.269289  0.178104   
1 -0.489017  0.052746 -0.640506 -0.941047 -0.056761  0.269289 -0.188684   
2 -0.884417  0.052746  0.975477 -0.363468 -0.056761 -1.996511  0.104747   

         V4        V5        V6        V7        V9        TP  Targets  
0 -0.382676  0.155328 -0.032119 -0.558132 -0.308204 -0.593742        0  
1 -0.718096  1.751069 -0.032119 -0.074237 -0.731282  0.938918        1  
2 -0.889739 -0.482968 -0.032119 -0.074237 -0.901180  1.398619        1

In [229]:
inputs = numerical_df4.iloc[:,:-1]
#dropping Targets column

# Logistic Regression using only numerical data

In [230]:
from sklearn.model_selection import train_test_split

In [231]:
x_train,x_test,y_train,y_test = train_test_split(inputs,targets,train_size = 0.8, random_state = 20)

In [232]:
print(x_train.shape,y_train.shape)
print(x_test.shape,y_test.shape)

(517764, 27) (517764, 1)
(129442, 27) (129442, 1)


In [233]:
from sklearn.linear_model import LogisticRegression
from sklearn import metrics

In [234]:
reg = LogisticRegression()

In [235]:
reg.fit(x_train,y_train)

/anaconda3/envs/py3-TF2.0/lib/python3.7/site-packages/sklearn/utils/validation.py:72: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(**kwargs)


LogisticRegression()

In [239]:
print('Score prediction:',reg.score(x_train,y_train).round(3)*100)

Score prediction: 84.3


We have split our main originial data into train and test. The model has an accuracy of $84.3%$ just accounting for numerical values. Now the goal is to one-hot encode categorical data and make predictions using it.

In [281]:
final = pd.concat([numerical_df4,dum_a0,dum_v8,dum_pm1,dum_c1,dum_c2],axis = 1)
final.head()

A1        A2        A3       PG1       PE1       PL1       PN1  \
0  0.540574 -0.206602  0.524361  0.432767  0.486076  0.023917 -0.310123   
1  1.931229  1.033375  1.052875 -1.233143 -1.445821  0.023917 -0.553836   
2 -1.010542 -0.301985 -0.052199 -1.425363 -1.445821  0.023917 -0.553836   
3  0.380113  0.318004  0.572408 -1.361290 -1.445821  0.023917 -0.797549   
4 -1.224489 -1.398888 -0.820946  2.226823  0.486076  0.023917 -1.284974   

        PO1        D0        D1        D2        D3        D4        D5  \
0  0.149802 -0.180638 -1.305960 -0.104297 -0.043828  0.646512 -0.035291   
1  0.149802 -0.180638 -0.647928 -0.104297 -0.043828 -1.405344 -0.701139   
2  0.149802 -0.180638  0.443019 -0.104297 -0.043828 -2.944236 -1.145037   
3  0.149802 -0.180638 -1.305960 -0.104297 -0.043828 -0.379416 -0.035291   
4  0.149802 -0.180638  0.564235 -0.104297 -0.043828  0.133548 -0.701139   

         D6       QTI        V0        V1       V10        V2        V3  \
0  2.344680  0.269630 -0.150814  2.091244 -0.056761  0.269289  0.178104   
1 -0.489017  0.052746 -0.640506 -0.941047 -0.056761  0.269289 -0.188684   
2 -0.884417  0.052746  0.975477 -0.363468 -0.056761 -1.996511  0.104747   
3 -0.027718  0.240712  0.387847  0.575098 -0.056761  0.269289 -0.115327   
4 -0.554917  0.269630 -1.081228 -1.085442 -0.056761  0.269289 -0.408757   

         V4        V5        V6        V7        V9        TP  Targets  \
0 -0.382676  0.155328 -0.032119 -0.558132 -0.308204 -0.593742        0   
1 -0.718096  1.751069 -0.032119 -0.074237 -0.731282  0.938918        1   
2 -0.889739 -0.482968 -0.032119 -0.074237 -0.901180  1.398619        1   
3 -0.331899  0.155328 -0.032119 -0.074237 -0.109436  1.081091        1   
4 -0.175274 -1.440413 -0.032119  0.167710 -0.140528 -0.658814        0   

   A0_East Anglia  A0_East Midlands  A0_Greater London  A0_Non-Mainland UK   \
0               0                 0                  0                    0   
1               0                 0                  0                    0   
2               0                 0                  0                    0   
3               0                 0                  0                    0   
4               1                 0                  0                    0   

   A0_North-East England  A0_North-West England  A0_Others  A0_Scotland  \
0                      0                      0          0            0   
1                      0                      0          0            0   
2                      0                      0          0            0   
3                      0                      0          0            0   
4                      0                      0          0            0   

   A0_South-East England  A0_South-West England  A0_Wales  A0_West Midlands  \
0                      0                      0         0                 1   
1                      0                      0         0                 0   
2                      1                      0         0                 0   
3                      0                      0         1                 0   
4                      0                      0         0                 0   

   A0_Yorks and Humbs  V8_At Home  V8_Car Park  V8_Garage  V8_Secure Carpark  \
0                   0           1            0          0                  0   
1                   1           1            0          0                  0   
2                   0           1            0          0                  0   
3                   0           1            0          0                  0   
4                   0           1            0          0                  0   

   PM1_A  PM1_B  PM1_C  PM1_D  PM1_M  PM1_N  PM1_O  PM1_P  PM1_S  PM1_W  \
0      0      0      0      0      1      0      0      0      0      0   
1      0      0      0      0      0      0      0      0      1      0   
2      0      0      0      0      0      0      0      0      1      0   
3      0      0      0      0

In [287]:
final.columns

Index(['A1', 'A2', 'A3', 'PG1', 'PE1', 'PL1', 'PN1', 'PO1', 'D0', 'D1', 'D2',
       'D3', 'D4', 'D5', 'D6', 'QTI', 'V0', 'V1', 'V10', 'V2', 'V3', 'V4',
       'V5', 'V6', 'V7', 'V9', 'TP', 'Targets', 'A0_East Anglia',
       'A0_East Midlands', 'A0_Greater London', 'A0_Non-Mainland UK ',
       'A0_North-East England', 'A0_North-West England', 'A0_Others',
       'A0_Scotland', 'A0_South-East England', 'A0_South-West England',
       'A0_Wales', 'A0_West Midlands', 'A0_Yorks and Humbs', 'V8_At Home',
       'V8_Car Park', 'V8_Garage', 'V8_Secure Carpark', 'PM1_A', 'PM1_B',
       'PM1_C', 'PM1_D', 'PM1_M', 'PM1_N', 'PM1_O', 'PM1_P', 'PM1_S', 'PM1_W',
       'C1_Comprehensive', 'C1_Third Party Fire and Theft',
       'C1_Third Party Only', 'C2_A', 'C2_B', 'C2_L', 'C2_M', 'C2_N', 'C2_O'],
      dtype='object')

In [288]:
final_order = ['A0_East Anglia',
       'A0_East Midlands', 'A0_Greater London', 'A0_Non-Mainland UK ',
       'A0_North-East England', 'A0_North-West England', 'A0_Others',
       'A0_Scotland', 'A0_South-East England', 'A0_South-West England',
       'A0_Wales', 'A0_West Midlands', 'A0_Yorks and Humbs', 'V8_At Home',
       'V8_Car Park', 'V8_Garage', 'V8_Secure Carpark', 'PM1_A', 'PM1_B',
       'PM1_C', 'PM1_D', 'PM1_M', 'PM1_N', 'PM1_O', 'PM1_P', 'PM1_S', 'PM1_W',
       'C1_Comprehensive', 'C1_Third Party Fire and Theft',
       'C1_Third Party Only', 'C2_A', 'C2_B', 'C2_L', 'C2_M', 'C2_N', 'C2_O','A1', 'A2', 'A3', 'PG1', 'PE1', 'PL1', 'PN1', 'PO1', 'D0', 'D1', 'D2',
       'D3', 'D4', 'D5', 'D6', 'QTI', 'V0', 'V1', 'V10', 'V2', 'V3', 'V4',
       'V5', 'V6', 'V7', 'V9', 'TP', 'Targets']

In [290]:
final = final[final_order]
final.head()
#This contains ALL data, combining categorical and numerical.
#GWP have been split into 0s and 1s base on the median to perform logistic regression.

A0_East Anglia  A0_East Midlands  A0_Greater London  A0_Non-Mainland UK   \
0               0                 0                  0                    0   
1               0                 0                  0                    0   
2               0                 0                  0                    0   
3               0                 0                  0                    0   
4               1                 0                  0                    0   

   A0_North-East England  A0_North-West England  A0_Others  A0_Scotland  \
0                      0                      0          0            0   
1                      0                      0          0            0   
2                      0                      0          0            0   
3                      0                      0          0            0   
4                      0                      0          0            0   

   A0_South-East England  A0_South-West England  A0_Wales  A0_West Midlands  \
0                      0                      0         0                 1   
1                      0                      0         0                 0   
2                      1                      0         0                 0   
3                      0                      0         1                 0   
4                      0                      0         0                 0   

   A0_Yorks and Humbs  V8_At Home  V8_Car Park  V8_Garage  V8_Secure Carpark  \
0                   0           1            0          0                  0   
1                   1           1            0          0                  0   
2                   0           1            0          0                  0   
3                   0           1            0          0                  0   
4                   0           1            0          0                  0   

   PM1_A  PM1_B  PM1_C  PM1_D  PM1_M  PM1_N  PM1_O  PM1_P  PM1_S  PM1_W  \
0      0      0      0      0      1      0      0      0      0      0   
1      0      0      0      0      0      0      0      0      1      0   
2      0      0      0      0      0      0      0      0      1      0   
3      0      0      0      0      0      0      0      0      1      0   
4      0      0      0      0      1      0      0      0      0      0   

   C1_Comprehensive  C1_Third Party Fire and Theft  C1_Third Party Only  C2_A  \
0                 1                              0                    0     0   
1                 1                              0                    0     0   
2                 1                              0                    0     0   
3                 1                              0                    0     0   
4                 1                              0                    0     0   

   C2_B  C2_L  C2_M  C2_N  C2_O        A1        A2        A3       PG1  \
0     0     1     0     0     0  0.540574 -0.206602  0.524361  0.432767   
1     0     0     0     0     0  1.931229  1.033375  1.052875 -1.233143   
2     0     0     0     0     0 -1.010542 -0.301985 -0.052199 -1.425363   
3     0     0     0     0     0  0.380113  0.318004  0.572408 -1.361290   
4     0     0     0     0     0 -1.224489 -1.398888 -0.820946  2.226823   

        PE1       PL1       PN1       PO1        D0        D1        D2  \
0  0.486076  0.023917 -0.310123  0.149802 -0.180638 -1.305960 -0.104297   
1 -1.445821  0.023917 -0.553836  0.149802 -0.180638 -0.647928 -0.104297   
2 -1.445821  0.023917 -0.553836  0.149802 -0.180638  0.443019 -0.104297   
3 -1.445821  0.023917 -0.797549  0.149802 -0.180638 -1.305960 -0.104297   
4  0.486076  0.023917 -1.284974  0.149802 -0.180638  0.564235 -0.104297   

         D3        D4        D5        D6       QTI        V0        V1  \
0 -0.043828  0.646512 -0.035291  2.344680  0.269630 -0.150814  2.091244   
1 -0.043828 -1.405344 -0.701139 -0.489017  0.052746 -0.640506 -0.941047   
2 -0.043828 -2.944236 -1.145037 -0.884417  0.052746  0.9

In [291]:
final.isna().sum()

A0_East Anglia           0
A0_East Midlands         0
A0_Greater London        0
A0_Non-Mainland UK       0
A0_North-East England    0
                        ..
V6                       0
V7                       0
V9                       0
TP                       0
Targets                  0
Length: 64, dtype: int64

In [292]:
final.to_csv('Last_preprocessing.csv', index=False)